docs/src/main/asciidoc/security-openid-connect-multitenancy.adoc

```diff
@@ -844,6 +844,73 @@ The default tenant configuration is automatically disabled when `quarkus.oidc.au
 
 Note that tenant specific configurations can also be disabled, for example: `quarkus.oidc.tenant-a.tenant-enabled=false`.
 
+== Authentication outside of the HTTP request context
+
+Sometimes it is practical to be able to create `SecurityIdentity` for given token when there is no active request context.
+The `quarkus-oidc` extension uses its own `OidcIdentityProvider` to convert a token to a `SecurityIdentity` instance.
+When the provider is used during an HTTP request, the tenant configuration can be resolved as described in the <<static-tenant-resolution>> section.
+However, when there is no active HTTP request, you need to select tenant explicitly with the `io.quarkus.oidc.TenantFeature` qualifier.
+One situation when you need to select the tenant configuration this way is when you are processing messages with the xref:vertx.adoc#event-bus[Vert.x event bus].
+In the example below, the 'product-order' message is consumed within different CDI request context, therefore injected `SecurityIdentity` would be anonymous.
+
+[source,java]
+----
+package org.acme.quickstart.oidc;
+
+import jakarta.annotation.security.RolesAllowed;
+import jakarta.inject.Inject;
+import jakarta.ws.rs.POST;
+import jakarta.ws.rs.Path;
+import org.eclipse.microprofile.jwt.JsonWebToken;
+import io.vertx.core.eventbus.EventBus;
+
+@Path("order")
+public class OrderResource {
+
+    @Inject
+    EventBus eventBus;
+
+    @Inject
+    JsonWebToken accessToken;
+
+    @RolesAllowed("customer")
+    @POST
+    public void order(String product) {
+        eventBus.publish("product-order", new Product(product, 1, accessToken.getRawToken()));
```

Comment:
Sorry, example is indeed good, but here we already have a verified token because the method is secure. So it does not describe the use case after all. I guess you need to drop `@RolesAllowed` and disable the proactive authentication, or perhaps, possibly in addition to the updated example, show a token passed as a custom header param, in which case you don't even need to deal with threads

Reply:
I'll just do it.

Reply:
I must admit my opinion slightly differ, IMO it's still fine that token has already been verified because you still need a way to convert token to the identity. Anyway, I believe I completely addressed this comment and I also added part about disabled proactive auth as you suggester.